src/data/ProgramsD2Repository.ts

```diff
@@ -101,92 +96,9 @@ export class ProgramsD2Repository implements ProgramsRepository {
-                        value: attribute.value,
-                        storedBy: attribute.storedBy,
-                    };
-                }),
-                programId: options.programId,
-            };
-        });
-    }
-
-    async saveAttributes(programs: ProgramAttributes[]): Async<Stats> {
-        if (programs.length === 0)
-            return { created: 0, ignored: 0, updated: 0, errorMessage: "", recordsSkipped: [] };
-        const teisToFetch = programs.map(program => program.id);
-        const programsIds = _(programs)
-            .map(program => program.programId)
-            .uniq()
-            .value();
-
-        const programsByKey = _(programs)
-            .keyBy(tei => tei.id)
-            .value();
-
-        const stats = await getInChunks<Stats>(teisToFetch, async teiIds => {
-            const trackedEntities = await this.getFromTracker<D2TrackedEntity>("trackedEntities", {
-                orgUnitIds: undefined,
-                programIds: programsIds,
-                trackedEntity: teiIds.join(";"),
-            });
-
-            const teisToSave = trackedEntities.map(tei => {
-                const currentProgram = programsByKey[tei.trackedEntity];
-                if (!currentProgram) throw Error(`Cannot find program: ${tei.trackedEntity}`);
-                const attributes = currentProgram?.attributes || tei.attributes;
-
-                return {
-                    trackedEntity: currentProgram?.id,
-                    orgUnit: tei.orgUnit,
-                    trackedEntityType: tei.trackedEntityType,
-                    attributes,
-                };
-            });
-
-            const response = await this.postTracker("trackedEntities", teisToSave);
-
-            return _(response)
-                .map(item => {
-                    const isError = item.status === "ERROR";
-                    return {
-                        created: item.stats.created,
-                        updated: item.stats.updated,
-                        ignored: item.stats.ignored,
-                        errorMessage: isError ? item.status : "",
-                        recordsSkipped: isError ? teisToSave.map(tei => tei.trackedEntity) : [],
-                    };
-                })
-                .value();
-        });
-
-        return stats.reduce((acum, stat) => {
-            return {
-                recordsSkipped: [...acum.recordsSkipped, ...stat.recordsSkipped],
-                errorMessage: `${acum.errorMessage}${stat.errorMessage}`,
-                created: acum.created + stat.created,
-                ignored: acum.ignored + stat.ignored,
-                updated: acum.updated + stat.updated,
-            };
-        }, initialStats);
-    }
-
     /* Private */
 
-    private async postTracker(key: TrackerDataKey, objects: object[]): Async<TrackerResponse[]> {
+    async postTracker(key: TrackerDataKey, objects: object[]): Async<TrackerResponse[]> {
```

Comment:
I see you made these methods public (and getFromTracker) so they can be used from the other repo. Typically, we would create a separate file (for example: D2Tracker.ts, with a class D2Tracker) that encapsulate this logic.